benchmark/dns/lookup-promises.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const common = require('../common.js');
+const { lookup } = require('dns').promises;
+
+const bench = common.createBenchmark(main, {
+  name: ['127.0.0.1', '::1'],
+  all: ['true', 'false'],
+  n: [5e6]
+});
+
+function main({ name, n, all }) {
+  var i = 0;
+
+  if (all === 'true') {
+    const opts = { all: true };
+    bench.start();
+    (async function cb() {
+      if (i++ === n) {
+        bench.end(n);
+        return;
+      }
+      await lookup(name, opts);
+      cb();
```

Comment:
If we’re using async/await anyway, would using a loop instead of recursive function calls make sense?

Reply:
Done. PTAL